src/shared/modules/dbMeta/dbMetaDuck.ts

```diff
@@ -128,6 +136,25 @@ export const getActiveDbName = (state: any) =>
  * Helpers
  */
 
+export const VERSION_FOR_EDITOR_HISTORY_SETTING = '4.3.0'
+
+export const versionHasEditorHistorySetting = (version: string) =>
+  semver.gte(version, VERSION_FOR_EDITOR_HISTORY_SETTING)
+
+export const supportsEditorHistorySetting = (state: any) =>
+  versionHasEditorHistorySetting(getVersion(state))
+
+export const shouldAllowOutgoingConnections = (state: any) =>
+  !isEnterprise(state) || getAllowOutgoingConnections(state)
+
+export const shouldRetainConnectionCredentials = (state: any) =>
+  !isEnterprise(state) || getRetainConnectionCredentials(state)
+
+export const shouldRetainEditorHistory = (state: any) =>
+  !isEnterprise(state) ||
+  !supportsEditorHistorySetting(state) ||
```

Comment:
If the setting is enterprise only, then it makes sense to more the enterprise check inside the supports editor history setting imo since community by definition doesn't support the setting

Reply:
Interesting idea about moving the check inside the supports function, I'll have a look at that.